tee-worker/omni-executor/rpc-server/src/methods/omni/get_message_code.rs

```diff
@@ -0,0 +1,34 @@
+use crate::server::RpcContext;
+use crate::ErrorCode;
+use executor_storage::{Storage, VerificationCodeStorage};
+use heima_authentication::web3::HeimaMessagePayload;
+use heima_identity_verification::helpers::generate_otp;
+use heima_primitives::{AccountId, Hashable};
+use jsonrpsee::{types::ErrorObject, RpcModule};
+use serde::Deserialize;
+use std::str::FromStr;
+
+#[derive(Deserialize)]
+pub struct GetMessageCodeParams {
+	pub omni_account: String,
+}
+
+pub fn register_get_message_code(module: &mut RpcModule<RpcContext>) {
+	module
+		.register_async_method("omni_getMessageCode", |params, ctx, _| async move {
+			let params = params.parse::<GetMessageCodeParams>()?;
+			let omni_account = AccountId::from_str(&params.omni_account).map_err(|_| {
+				log::error!("Could not parse AccountId: {:?}", params.omni_account);
+				ErrorCode::InvalidParams
+			})?;
+			let verification_code_storage = VerificationCodeStorage::new(ctx.storage_db.clone());
+			let message_code = generate_otp(8);
+
+			verification_code_storage
+				.insert(&omni_account.hash(), message_code.clone())
```

Comment:
My two concerns here:
* this is an only growing storage (old codes are not pruned - can be only overriden - it's relatively cheap for somebody to flood worker with prepared requests  - I haven't measured the impact)
* people can override code for each other

Reply:
> this is an only growing storage (old codes are not pruned - can be only overriden - it's relatively cheap for somebody to flood worker with prepared requests - I haven't measured the impact)

This is a good point. I'll remove the entry after the code is verified.

> people can override code for each other

Should we make this idempotent until the code is used? If there is a code in storage we just return it instead of inserting a new one?